tests/Integration/Schema/Directives/InheritsDirectiveTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Tests\Integration\Schema\Directives;
+
+use Tests\TestCase;
+
+class InheritsDirectiveTest extends TestCase
+{
+    public function testSimpleInheritance(): void
+    {
+        $schema = $this->buildSchema(/**@lang GraphQL */ '
+            type ParentType {
+                attribute_1: String
+            }
+
+            type ChildType @inherits(from: ParentType) {
+                attribute_2: String
+            }
+        ' . self::PLACEHOLDER_QUERY);
+
+        $childType = $schema->getType('ChildType');
+
+        $this->assertNotNull($childType->getField('attribute_1'));
+    }
+
+    public function testInheritTransitive(): void
+    {
+        $schema = $this->buildSchema(/**@lang GraphQL */ '
+            type GrandParentType {
+                attribute_1: String
+            }
+
+            type ParentType @inherits(from: GrandParentType) {
+                attribute_2: String
+            }
+
+            type ChildType @inherits(from: ParentType) {
+                attribute_3: String
+            }
+        ' . self::PLACEHOLDER_QUERY);
+
+        $childType = $schema->getType('ChildType');
+
+        $this->assertNotNull($childType->getField('attribute_1'));
+        $this->assertNotNull($childType->getField('attribute_2'));
+        $this->assertNotNull($childType->getField('attribute_3'));
+    }
+
+    public function testSchemaOrderIsInsignificant(): void
```

Comment:
This test highlights a problem with the current implementation. Given that the schema can be spread across multiple different files, the order of type definitions might not be predictable in every case. Regardless of definition order, inheritance should always lead to the same result.

There could be other solutions to this, but my instinct is to recursively resolve the inheritance of parent types before merging the attributes into a type.